src/plugins/intel_cpu/src/nodes/if.cpp

```diff
@@ -17,31 +19,43 @@ namespace ov {
 namespace intel_cpu {
 namespace node {
 
+inline size_t getElementNum(const MemoryPtr& srcMemPtr) {
+    auto srcDims = srcMemPtr->getStaticDims();
+    return std::accumulate(srcDims.begin(), srcDims.end(), 1, std::multiplies<size_t>());
+}
+
 If::PortMapHelper::PortMapHelper(const MemoryPtr &from, const std::deque<MemoryPtr>& to,
                                            const dnnl::engine& eng) : srcMemPtr(from), dstMemPtrs(to) {
     size = 0;
     if (srcMemPtr->getDesc().isDefined())
-        size = srcMemPtr->getSize();
+        size = getElementNum(srcMemPtr);
 }
 
 void If::PortMapHelper::execute(dnnl::stream& strm) {
     // if output shapes are changed,
     // after subgraph inference we should redefine out memory of 'If'
     redefineTo();
 
-    cpu_memcpy(dstMemPtrs.front()->getData(), srcMemPtr->getData(), size);
+    cpu_convert(srcMemPtr->getData(),
+                dstMemPtrs.front()->getData(),
+                srcMemPtr->getDesc().getPrecision(),
+                dstMemPtrs.front()->getDesc().getPrecision(),
+                size);
 }
 
 void If::PortMapHelper::redefineTo() {
     const auto &currDesc = dstMemPtrs.front()->getDesc();
     if (currDesc.getShape().isDynamic() || currDesc.getShape().getStaticDims() != srcMemPtr->getStaticDims()) {
         // TODO : check the entire dstMemPtrs usage considering the proper memory sharing
-        auto memDesc = srcMemPtr->getDescPtr();
+        auto newShape = srcMemPtr->getDesc().getShape().getDims();
```

Comment:
Maybe I missed something, but we can just call `srcMemPtr->getStaticDims()` there, cannot we? Correct me please if I'm wrong.

Reply:
Updated